### PR TITLE
AWS cleanup via aws-cleanup

### PR DESCRIPTION
**Pull request by OdedViner**
Destroy AWS cluster base on cluster name
```
aws-cleanup --cluster-name <cluster-name> --force
```
Signed-off-by: OdedViner <oviner@redhat.com>

### CODE REVIEW

File: ocs_ci/cleanup/aws/cleanup.py [hunk 1]
```diff
@@ -129,7 +129,9 @@ def cleanup(cluster_name, cluster_id, upi=False, failed_deletions=None):
     delete_cluster_buckets(cluster_name)
 
 
-def get_clusters(time_to_delete, region_name, prefixes_hours_to_spare):
+def get_clusters(
+    time_to_delete, region_name, prefixes_hours_to_spare, cluster_pattern=None
```
Comment (petr-balogh) [inline ocs_ci/cleanup/aws/cleanup.py:133]: cluster_pattern is not documented in docstring

File: ocs_ci/cleanup/aws/cleanup.py [hunk 2]
```diff
@@ -181,13 +183,31 @@ def determine_cluster_deletion(ec2_instances, cluster_name):
                         return True
         return False
 
+    def determine_cluster_deletion_base_name(ec2_instance_objs, vpc_id):
+        # Get all instances
```
Comment (petr-balogh) [inline ocs_ci/cleanup/aws/cleanup.py:187]: Please add docstring

File: ocs_ci/cleanup/aws/cleanup.py [hunk 3]
```diff
@@ -181,13 +183,31 @@ def determine_cluster_deletion(ec2_instances, cluster_name):
...
+        vpc_ids = []
+        for ec2_instance in ec2_instance_objs:
+            vpc_ids.append(ec2_instance.get("Instances")[0].get("VpcId"))
```
Comment (petr-balogh) [inline ocs_ci/cleanup/aws/cleanup.py:190]:
```suggestion
        vpc_ids = [ec2_instance.get("Instances")[0].get("VpcId") for ec2_instance in ec2_instance_objs]
```

File: ocs_ci/cleanup/aws/cleanup.py [hunk 4]
```diff
@@ -272,13 +296,14 @@ def cluster_cleanup():
...
-        description="AWS overall resources cleanup according to running time"
+        description="AWS overall resources cleanup according to running time",
+        formatter_class=argparse.RawTextHelpFormatter,
     )
     parser.add_argument(
         "--hours",
         type=hour_valid,
         action="store",
-        required=True,
+        required=False,
```
Comment (petr-balogh) [inline ocs_ci/cleanup/aws/cleanup.py:306]: @clacroix12 do you remember, can this somehow affect our current flow of AWS cleanup script?

File: ocs_ci/cleanup/aws/cleanup.py (hunk 3 above)
Comment (OdedViner) [reply ocs_ci/cleanup/aws/cleanup.py:190]: done

File: ocs_ci/cleanup/aws/cleanup.py (hunk 1 above)
Comment (OdedViner) [reply ocs_ci/cleanup/aws/cleanup.py:133]: done

File: ocs_ci/cleanup/aws/cleanup.py (hunk 2 above)
Comment (OdedViner) [reply ocs_ci/cleanup/aws/cleanup.py:187]: done

File: ocs_ci/cleanup/aws/cleanup.py (hunk 4 above)
Comment (OdedViner) [reply ocs_ci/cleanup/aws/cleanup.py:307]: ping:)

**Review by clacroix12 — COMMENTED** [review]
I don't see any glaring issues but I think it would be smart to run our standard aws-cleanup with this branch to verify that everything is working as intended for that code path.

**Review by petr-balogh — APPROVED** [review]

**Comment by petr-balogh** [comment]
@clacroix12 can you please review/approve?

**Comment by OdedViner** [comment]
PR validation pass https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/view/all/job/qe-aws-cleanup/16766/console

**Pull request merged**